codefresh.yml

```diff
@@ -1,58 +1,62 @@
 version: "1.0"
+stages:
+  - setup
+  - build
+  - finalise
 steps:
-  TagReleaseName:
-    title: Setup RELEASE_NAME variable with tag
-    image: alpine/git
-    when:
-      condition:
-        all:
-          releaseNameNotDefined: "${{TAG_REPO}} == true"
-    commands:
-      - cf_export RELEASE_NAME=${{TAG_NAME}}
+  main_clone:
+    title: Cloning main repository...
+    stage: 'setup'
+    type: git-clone
+    git: github
+    repo: "${{CF_REPO_OWNER}}/${{CF_REPO_NAME}}"
+    revision: "${{CF_REVISION}}"
 
-  BranchReleaseName:
-    title: Setup RELEASE_NAME variable with branch name
+  SetProperties:
+    title: Setup build
+    stage: 'setup'
     image: alpine/git
-    when:
-      condition:
-        all:
-          releaseNameNotDefined: "${{TAG_REPO}} == false"
     commands:
-      - cf_export RELEASE_NAME=${{GLOBAL_RELEASE_VERSION}}.${{CF_BRANCH_TAG_NORMALIZED}}
+      - cf_export CICD_REPO="${{CF_VOLUME_PATH}}/m2_repository"
+      - cf_export GPG_IMPORT_COMMAND="cat <(echo -e '${{GPG_PRIVATE_KEY}}') | gpg --batch --import"
+      - cf_export MAVEN_BUILD_PROFILES="regnosys"
 
   ReleaseProperties:
     title: Setup release build
+    stage: 'setup'
     image: alpine/git
     when:
       condition:
         all:
           releaseNameNotDefined: "${{TAG_REPO}} == true"
     commands:
-      - cf_export MVN_SET_VERSION="versions:set-property -DnewVersion=${{RELEASE_NAME}}"
+      - cf_export RELEASE_NAME=${{TAG_NAME}}
 
-  BranchProperties:
-    title: Setup branch build
+  SnapshotProperties:
+    title: Setup snapshot build
+    stage: 'setup'
     image: alpine/git
     when:
       condition:
         all:
           releaseNameNotDefined: "${{TAG_REPO}} == false"
     commands:
-      - cf_export MVN_SET_VERSION="versions:update-property -DnewVersion=[${{RELEASE_NAME}}] -DallowSnapshots=true -DallowDowngrade=true"
+      - cf_export RELEASE_NAME=${{GLOBAL_RELEASE_VERSION}}.${{CF_BRANCH_TAG_NORMALIZED}}-SNAPSHOT
 
   Build:
     title: Maven build
-    fail_fast: true
-    image: regnosys-docker-registry.jfrog.io/regnosys/rosetta-products-deps:java-17
+    stage: 'build'
+    fail_fast: false
+    image: maven:3.8.5-openjdk-17
     working_directory: ./
     commands:
-      - rm -rf /codefresh/volume/.m2/com/regnosys
-      - mvn -s /settings.xml versions:set -DnewVersion=${{RELEASE_NAME}}
-      - mvn -s /settings.xml ${{MVN_SET_VERSION}} -Dproperty=regnosys.bundle.version
```

Comment:
we are actually running the `versions` plugin a second time here - using the configured variable. But you haven't brought this across.